### PR TITLE
New UTs, and split out into separate UT files.

### PR DESCRIPTION
**Pull request by robbrockbank**

### CODE REVIEW

File: calico_containers/tests/unit/checksystem_test.py [hunk 1]
```diff
@@ -69,3 +71,99 @@ def m_module_loaded(module):
...
+        :param docker_version_status: return_value for _check_docker_version
+        """
+        with \
+                patch('calico_ctl.checksystem.enforce_root', autospec=True) \
+                        as m_enforce_root, \
+                patch('calico_ctl.checksystem._check_kernel_modules', autospec=True) \
+                        as m_check_kernel_modules, \
+                patch('calico_ctl.checksystem._check_ip_forwarding', autospec=True) \
+                        as m_check_ip_forwarding, \
+                patch('calico_ctl.checksystem._check_docker_version', autospec=True) \
```
Comment (TrimBiggs) [inline calico_containers/tests/unit/checksystem_test.py]: This with would be cleaner with parens, if possible.

File: calico_containers/tests/unit/checksystem_test.py [hunk 2]
```diff
@@ -69,3 +71,99 @@ def m_module_loaded(module):
...
+        Test for check_system when one of the system checks fails
+
+        This test does not quit if there is an error -
+        Assert that the function returns False
+
+        :param kernel_status: return_value for _check_kernel_modules
+        :param ip_forwarding_status: return_value for _check_ip_forwarding
+        :param docker_version_status: return_value for _check_docker_version
+        """
+        with \
```
Comment (TrimBiggs) [inline calico_containers/tests/unit/checksystem_test.py]: this first \ isn't really necessary

File: calico_containers/tests/unit/checksystem_test.py [hunk 3]
```diff
@@ -69,3 +71,99 @@ def m_module_loaded(module):
...
+        :param docker_version_status: return_value for _check_docker_version patch
+        """
+        with \
+                patch('calico_ctl.checksystem.enforce_root', autospec=True) \
+                        as m_enforce_root, \
+                patch('calico_ctl.checksystem._check_kernel_modules', autospec=True) \
+                        as m_check_kernel_modules, \
+                patch('calico_ctl.checksystem._check_ip_forwarding', autospec=True) \
+                        as m_check_ip_forwarding, \
+                patch('calico_ctl.checksystem._check_docker_version', autospec=True) \
```
Comment (TrimBiggs) [inline calico_containers/tests/unit/checksystem_test.py]: Same as above for this `with` statement

File: calico_containers/tests/unit/checksystem_test.py [hunk 4]
```diff
@@ -69,3 +71,99 @@ def m_module_loaded(module):
...
+            # Set up mock objects
+            m_check_kernel_modules.return_value = kernel_status
+            m_check_ip_forwarding.return_value = ip_forwarding_status
+            m_check_docker_version.return_value = docker_version_status
+
+            # Call method under test expecting a SystemExit when fail detected
+            self.assertRaises(SystemExit, check_system, fix=False, quit_if_error=True)
+
+            # Assert
+            m_enforce_root.assert_called_once_with()
```
Comment (TrimBiggs) [inline calico_containers/tests/unit/checksystem_test.py]: When I first suggested editing this file, all of the `check_kernel_modules`, `check_ip_forwarding`, and `check_docker_version` function code were within the `check_system`function.  These functions are the real meat of the check_system file that should be tested.  We should add tests for these or put it on the backlog.

File: calico_containers/tests/unit/diags_test.py [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+            call().__enter__().writelines(m_sh_command_return()),
+            call(diags_dir + '/iptables', 'w'),
+            call().__enter__().writelines(m_sh_command_return()),
+            call(diags_dir + '/ipset', 'w'),
+            call().__enter__().writelines(m_sh_command_return()),
+            call(diags_dir + '/etcd_calico', 'w'),
+            call().__enter__().write('dir?, key, value\n'),
+            call().__enter__().write('DIR,  666,\n'),
+            call().__enter__().write('FILE, 555, 999\n')
+        ], any_order=True)
```
Comment (TrimBiggs) [inline calico_containers/tests/unit/diags_test.py]: Seems like the save_diags function is pretty large.  Should we have some helper functions instead that can be called to slim down the function for things like writing netstat, writing routes, writing iptables, etc...

File: calico_containers/tests/unit/endpoint_test.py [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+# limitations under the License.
+
+import unittest
+
+from mock import patch
+from nose_parameterized import parameterized
+from calico_ctl import endpoint
+
+
+class TestEndpoint(unittest.TestCase):
```
Comment (TrimBiggs) [inline calico_containers/tests/unit/endpoint_test.py]: Looks like Walker didn't get to actually writing any of the tests for the endpoint command calls

File: calico_containers/tests/unit/pool_test.py [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+import unittest
+
+from mock import patch
+from nose_parameterized import parameterized
+
+from calico_ctl import pool
+
+
+class TestPool(unittest.TestCase):
```
Comment (TrimBiggs) [inline calico_containers/tests/unit/pool_test.py]: Looks like Walker didn't get to writing the pool command calls here also.

File: calico_containers/tests/unit/profile_test.py [hunk 8]
```diff
@@ -0,0 +1,432 @@
...
+    ])
+    def test_profile_rule_add_remove_remove_inbound(self, direction_arg,
+                                                    position_arg):
+        """
+        Test for profile_rule_add_remove function when removing a Rule from a
+        Profile
+
+        Test both direction - inbound and outbound
+        Test positions None and 1
+        (if out of bounds positions provided, function will not remove the Rule)
```
Comment (TrimBiggs) [inline calico_containers/tests/unit/profile_test.py]: Not sure why this is in parentheses

File: calico_containers/tests/unit/profile_test.py [hunk 9]
```diff
@@ -0,0 +1,432 @@
...
+        self.assertRaises(SystemExit, profile_rule_add_remove,
+                          operation, name, position, action, direction,
+                          protocol='icmp', src_ports=[40,60])
+
+    @patch('calico_ctl.profile.client', autospec=True)
+    def test_profile_rule_add_remove_fail_get_profile(self, m_client):
+        """
+        Test for profile_rule_add_remove when client cannot obtain a profile
+
+        Assert that the system exits
```
Comment (TrimBiggs) [inline calico_containers/tests/unit/profile_test.py]: Most test function docstrings do not have periods.

File: calico_containers/tests/unit/profile_test.py [hunk 10]
```diff
@@ -0,0 +1,432 @@
...
+            new_rule = Rule(**rule_dict)
+
+            # Call method under test
+            profile_rule_add_remove(operation, name, position, action, direction,
+                                    icmp_type='5', icmp_code='5')
+
+            # Assert
+            m_client.get_profile.assert_called_once_with(name)
+            m_client.profile_update_rules.assert_called_once_with(m_Profile)
+            if direction_arg=='inbound':
```
Comment (TrimBiggs) [inline calico_containers/tests/unit/profile_test.py]: Spacing

File: calico_containers/tests/unit/profile_test.py [hunk 11]
```diff
@@ -0,0 +1,432 @@
...
+            m_Profile = Mock(spec=Profile, name=name, rules=m_Rules)
+            m_client.get_profile.return_value = m_Profile
+
+            # Call method under test
+            profile_rule_add_remove(operation, name, position, action, direction)
+
+            # Assert
+            m_client.get_profile.assert_called_once_with(name)
+            m_client.profile_update_rules.assert_called_once_with(m_Profile)
+            if direction_arg=='inbound':
```
Comment (TrimBiggs) [inline calico_containers/tests/unit/profile_test.py]: Here too

File: calico_containers/tests/unit/diags_test.py (hunk 5 above)
Comment (robbrockbank) [inline calico_containers/tests/unit/diags_test.py]: added to backlog

File: calico_containers/tests/unit/checksystem_test.py (hunk 4 above)
Comment (robbrockbank) [inline calico_containers/tests/unit/checksystem_test.py]: general improve coverage added to backlog

File: calico_containers/tests/unit/checksystem_test.py (hunk 3 above)
Comment (robbrockbank) [inline calico_containers/tests/unit/checksystem_test.py]: got rid of additional \

File: calico_containers/tests/unit/endpoint_test.py (hunk 6 above)
Comment (robbrockbank) [inline calico_containers/tests/unit/endpoint_test.py]: backlog

File: calico_containers/tests/unit/pool_test.py (hunk 7 above)
Comment (robbrockbank) [inline calico_containers/tests/unit/pool_test.py]: backlog

File: calico_containers/tests/unit/profile_test.py (hunk 8 above)
Comment (robbrockbank) [inline calico_containers/tests/unit/profile_test.py]: i think just to indicate it's explicitly not testing it.  i'll tweak it.

File: calico_containers/tests/unit/profile_test.py (hunk 9 above)
Comment (robbrockbank) [inline calico_containers/tests/unit/profile_test.py]: We can fix these up when we next work on the files.  I'll fix up the ones I see.

**Comment by robbrockbank** [comment]
Code markups applied.  Merging.

**Pull request merged**